docs/guides/migrations.md

```diff
@@ -0,0 +1,83 @@
+<!--[meta]
+section: guides
+title: Migrations
+[meta]-->
+
+## Migrations in Keystone
+
+Migrations are required when you update a List in Keystone after the database has been initialised. **Keystone can't automatically manage these migrations for you** because it doesn't know what to do with the existing data. This guide will explain when and why you need database migrations and offer some basic solutions to get you started. As your project grows to include production deployments, how you manage the seeding and migration of data, will depend on your specific situation. Keystone tries not to limit type of applications you build and therefor cannot offer a "one size fits all" solution to migrations.
+
+When you create a new List, depending on the adapter, Keystone will initialise a database table with columns for each of the fields defined in the List. Let's look at an example.
+
+```javascript
+const { KnexAdapter } = require('@keystonejs/adapter-knex');
+
+const keystone = new Keystone({
+  name: 'KnexMigrationDemo',
+  adapter: new KnexAdapter(),
+});
+
+keystone.createList('ExampleList', {
+  fields: {
+    age: { type: Integer },
+  },
+});
+```
+
+Using the KnexAdapter if we create an `ExampleList` that contains an `Integer` field called `age` when we start Keystone for the first time it will generate a table with an `id` and `age` column:
+
+| table_name  | column_name | data_type |
+| ----------- | ----------- | --------- |
+| ExampleList | id          | integer   |
+| ExampleList | age         | integer   |
+
+If we update our Keystone List either by renaming fields or adding new fields, we might encounter errors next time we start Keystone. This is because the database schema has changed and Keystone doesn't know what to do with existing data or won't be able to save new data. Let's imagine that because we are responsible with data collection we want to update the list to only include an age range rather than a specific age. We might update the field to use a select type:
+
+```javascript
+keystone.createList('MigrationExample', {
+  fields: {
+    age: { type: Select, options: '0-18, 19-35, 36-50, 50+' },
+  },
+});
+```
+
+If we restart Keystone after this change it will throw an error. The `data_type` of the `age` column is now a `text` value and it should contain a string representation of the age range. The table should be change to something like the following:
```

Comment:
```suggestion
If we restart Keystone after this change it will throw an error. The `data_type` of the `age` column is now a `text` value and it should contain a string representation of the age range. The table should change to something like the following:
```